src/webui/api/torrentscontroller.cpp

```diff
@@ -1038,3 +1038,78 @@ void TorrentsController::categoriesAction()
 
     setResult(categories);
 }
+
+void TorrentsController::addTagsAction()
+{
+    checkParams({"hashes", "tags"});
+
+    const QStringList hashes {params()["hashes"].split('|')};
+    const QStringList tags {params()["tags"].split(',', QString::SkipEmptyParts)};
+
+    for (const auto &tagRaw : asConst(tags)) {
+        const QString tag {tagRaw.trimmed()};
+        applyToTorrents(hashes, [tag](BitTorrent::TorrentHandle *const torrent)
+        {
+            if (!torrent->hasTag(tag) && !torrent->addTag(tag))
+                throw APIError(APIErrorType::Conflict, tr("Incorrect tag name"));
```

Comment:
Maybe check tag name once before applying it to torrents?

Reply:
It's checked in `TorrentHandle::addTag` anyway.